Fix #200 Fix #236

mattben opened the pull request.

mattben commented:
@James-Crean um... I broke it

James-Crean commented:
cd into frontend and then run `npm test`

That will give you the test output so you can look at where things went wrong. 

If you find you need to edit and debug test code, the [https://github.com/UV-CDAT/vcdat/tree/master/frontend/test/mocha#writing-tests](README.md) mentions a few steps that will make things easier. (especially the debugging tests part)

James-Crean commented on frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx:
```diff
@@ -244,34 +245,70 @@ class FileTab extends Component {
-            bookmarks.push(file)
+            let fileInBookmarks = false
+            let fileInBookmarksIndex = -1
+            bookmarks.reduce((prev_val, current_obj) => {
```
May need to check the logic around here. Adding bookmarks doesnt always work and removing often removes the wrong bookmark.

mattben replied:
:ballot_box_with_check:

mattben commented on .gitignore:
```diff
@@ -19,5 +19,6 @@ backend/vcdat/resources
 build/
 dist/
 *.egg-info
+frontend/package-lock.json
```
agh, if there's a reason its good enough for me I'll add it back next week.

mattben commented on frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx:
```diff
@@ -244,34 +245,70 @@ class FileTab extends Component {
-    handleDragEnd(){
-        this.setState({showBookmarkZone: false})
+    handleDragEnd(event, option){
+        console.log("drag end " + option)
```
:ballot_box_with_check:

mattben commented on frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx:
```diff
@@ -244,34 +245,70 @@ class FileTab extends Component {
+            this.setState({showRemoveBookmarkZone: false})
+        }
+        else{
+            console.log('Error')
```
:ballot_box_with_check:

mattben commented:
@James-Crean (@doutriaux1) I addressed all the requested changes, and corrected the test. I added a new parameter, that the test was not passing. Once this passes it should be safe to merge.
